### DISCUSSION TITLE
Reading the output .npy files

### DISCUSSION THREAD

**Discussion by pigaral** [mboquien/cigale#90] [Q&A] [answered]
I'm interested in reading the .npy files that are spit out by cigale when save_chi2 is set to properties. I can run the command pcigale-plots pdf and pcigale-plots chi2 without issue, but when I try to load in the .npy files in a jupyter-notebook with np.load(/path/to/file), I get an error:

> ValueError: Cannot load file containing pickled data when allow_pickle=False

When I then set allow_pickle=True:

> UnpicklingError: Failed to interpret file 'CIGALE/J0224_dust/out/104_stellar.m_star_chi2-block-0.npy' as a pickle

I read in the cigale documentation that there should be a read_chi.py file somewhere, but searching my entire computer system came back empty. 

My goal of reading this file is to obtain the asymmetric errorbars on the bayesian variables. Would someone be able to advise the best way to do so, or advise the best way to read the output .npy files so that I can calculate those values myself?

Using the 2022.1 version on the code on Ubuntu 20.04 LTS

Thank you in advance!

**Comment by mboquien** [comment, marked as answer]
With the 2022.1 version, these are actually memmap'ed files. You can see how they are read on [lines 69-70 of pcigale_plots/plot_types/\_\_init\_\_.py](https://gitlab.lam.fr/cigale/cigale/-/blob/v2022.1/pcigale_plots/plot_types/pdf/__init__.py?ref_type=tags#L69). With the latest development version ([`develop` branch on the git repository](https://gitlab.lam.fr/cigale/cigale/-/tree/develop?ref_type=heads)), I have changed them to simple FITS files so they are easier to handle.

**Reply by pigaral** [reply]
Works perfectly! Thank you for your help!
